SteamKit2/SteamKit2/Steam/CDN/Client.cs

```diff
@@ -81,11 +83,60 @@ public async Task<DepotManifest> DownloadManifestAsync( uint depotId, ulong mani
                 url = $"depot/{depotId}/manifest/{manifestId}/{MANIFEST_VERSION}";
             }
 
-            var manifestData = await DoRawCommandAsync( server, url, proxyServer ).ConfigureAwait( false );
+            using var request = new HttpRequestMessage( HttpMethod.Get, BuildCommand( server, url, proxyServer ) );
 
-            manifestData = ZipUtil.Decompress( manifestData );
+            using var cts = new CancellationTokenSource();
+            cts.CancelAfter( RequestTimeout );
+
+            DepotManifest depotManifest;
+
+            try
+            {
+                using var response = await httpClient.SendAsync( request, HttpCompletionOption.ResponseHeadersRead, cts.Token ).ConfigureAwait( false );
+
+                if ( !response.IsSuccessStatusCode )
+                {
+                    throw new SteamKitWebRequestException( $"Response status code does not indicate success: {response.StatusCode:D} ({response.ReasonPhrase}).", response );
+                }
+
+                if ( !response.Content.Headers.ContentLength.HasValue )
+                {
+                    throw new SteamKitWebRequestException( "Response does not have Content-Length", response );
+                }
+
+                cts.CancelAfter( ResponseBodyTimeout );
+
+                var contentLength = ( int )response.Content.Headers.ContentLength;
+                var buffer = ArrayPool<byte>.Shared.Rent( contentLength );
+
+                try
+                {
+                    using var ms = new MemoryStream( buffer, 0, contentLength );
+
+                    // Stream the http response into the rented buffer
+                    await response.Content.CopyToAsync( ms, cts.Token );
+
+                    ms.Position = 0;
+
+                    // Decompress the zipped manifest data
+                    using var zip = new ZipArchive( ms );
```

Comment:
ZipArchive needs a seekable stream fyi (otherwise it literally copies the stream into its own temp MemoryStream).